### PR TITLE
theme presets, import and export, better theme gui

### PR DESCRIPTION
**Pull request by Peter-Crawley**

### CODE REVIEW

File: workshop/controllers/ui/components/themePreviewer.lua [hunk 1]
```diff
@@ -29,16 +136,16 @@ for _,prop in pairs(themer.types) do
...
-   local size = 1/count
-   local i = 0
+   size = 1/count
+   i = 0
 
    for k,v in pairs(theme[prop]) do
       if type(v) == "colour" then
-         local ch,cs,cv = v:getHSV()
-         local btn = ui.create("guiTextBox", themeProperty, {
+         ch,cs,cv = v:getHSV()
```
Comment (teverse) [inline workshop/controllers/ui/components/themePreviewer.lua]: why is this done?

File: workshop/controllers/ui/components/themePreviewer.lua [hunk 2]
```diff
@@ -65,5 +173,78 @@ for _,prop in pairs(themer.types) do
...
+function canvasSet(size)
+   container.parent.canvasSize = size
+   container.size = size
+end
+
+function makePresetMenu()
+   preset = shared.workshop:getSettings("themeType")
+   THISISTHERESETBUTTON.visible = false
+   customUI.visible = false
+   _, _ = pcall(canvasSet, guiCoord(1, 0, 1, 0))
```
Comment (teverse) [inline workshop/controllers/ui/components/themePreviewer.lua]: why is this assigned to _, _?

File: workshop/controllers/ui/components/themePreviewer.lua [hunk 3]
```diff
@@ -65,5 +173,78 @@ for _,prop in pairs(themer.types) do
...
+   if shared.workshop:getSettings("themeType") == "custom" then
+      THISISTHERESETBUTTON.visible = true
+      customUI.visible = true
+      _, _ = pcall(canvasSet, guiCoord(1, 0, 0, 560))
```
Comment (teverse) [inline workshop/controllers/ui/components/themePreviewer.lua]: why are the return values assigned to _?

File: workshop/controllers/ui/components/themePreviewer.lua (hunk 1 above)
Comment (Peter-Crawley) [reply workshop/controllers/ui/components/themePreviewer.lua]: if you look its still there, its just not a local varible anymore and that is triping it up

Comment (Peter-Crawley) [reply workshop/controllers/ui/components/themePreviewer.lua]: everything still works

Comment (teverse) [reply workshop/controllers/ui/components/themePreviewer.lua]: I mean, I don't see these three variables being used anywhere?

File: workshop/controllers/ui/components/themePreviewer.lua (hunk 3 above)
Comment (Peter-Crawley) [reply workshop/controllers/ui/components/themePreviewer.lua]: during workshop loading i was getting some crashes from the code that function contains, this function is a case of, its nice, if it works, good, if not, no big deal

File: workshop/controllers/ui/components/themePreviewer.lua (hunk 1 above)
Comment (Peter-Crawley) [reply workshop/controllers/ui/components/themePreviewer.lua]: i was having issues with local and non local varibles, I could not find the ones causing problems, so I set them all to global.

**Pull request merged**